src/main/java/com/app/katchup/DistributedCalendar/CalendarController.java

```diff
@@ -25,53 +28,46 @@
 public class CalendarController {
 
     private static final Logger logger = LogManager.getLogger(CalendarController.class);
-
     @Autowired
     CalendarService calendarService;
     @Autowired
     UserService userService;
     @Autowired
     MeetingService meetingService;
-
+    MeetingResponseRepository meetingResponseRepository;
 
     @GetMapping("/{userName}")
     public ResponseEntity<DistributedCalendar> showUserEvents(@PathVariable String userName, HttpServletRequest request) {
-
-        User user = userService.getUserByUserName(userName);
-        if (user == null) {
-            logger.info("user name is not valid");
+        if (!(userService.isCredentialsMatched(userName, request.getHeader("password"))))
             return new ResponseEntity<>(HttpStatus.UNAUTHORIZED);
-        }
 
-        boolean t = userService.isCredentialsMatched(user.getUserName(), request.getHeader("password"));
-        if (t) {
-            logger.info("user password not valid");
-            return new ResponseEntity<>(HttpStatus.UNAUTHORIZED);
-        } else {
-            List<String> meetingIds = calendarService.getAcceptedMeetingIds(user.getUserName());
+        else {
+            List<MeetingID> meetingIds = calendarService.getAcceptedMeetingIds(userName);
+            List<Meeting> meetingList = new ArrayList<>();
 
-            List<Meeting> meetingList = meetingService.getMeetingDetailsForMeetingIds(meetingIds);
+            if (meetingIds.size() <= 0)
+                throw new UnAuthorizedException("Sorry! User is not allowed to enter this meeting");
```

Comment:
this exception doesnt mean that there is not authorization issue. there could be a reason that the user didnt accept any meetings. if that is the case, return an empty object [] with ok